charts/render.py

```diff
@@ -166,6 +166,23 @@ def replace_magic_strings(rendered_yaml):
     return result
 
 
+def remove_prometheus_rules(template, drop_rules):
+    groups = []
+
+    for group in template.get('spec', {}).get('groups', []):
+        if group.get('name') in drop_rules:
+            for rule in group.get('rules', [])[:]:
+                if rule.get('alert') in drop_rules[group['name']] \
+                       or rule.get('record') in drop_rules[group['name']]:
+                    group['rules'].remove(rule)
+        if group.get('rules'):
+            groups.append(group)
+
+    template['spec']['groups'] = groups
+
+    return template
```

Comment:
Haha that's what I was doing at the beginning, but though it would be more efficient to directly edit the dict, that's why there is still this return here which is useless for the moment. Will change this.

Reply:
Efficiency really isn't a concern here :P Whereas non-mutating methods are easier to test ;)